### ISSUE TITLE
`Host function not implemented` when forking v1.11 chain. 

### ISSUE THREAD

**Issue by metricaez** [AcalaNetwork/chopsticks#754]
We are currently facing an issue when forking a chain that is at v1.11 with the following log: 

```
[12:09:46.874] INFO (block-builder): Failed to apply extrinsic {} Error: Host function not implemented: ext_storage_proof_size_storage_proof_size_version_1
    app: "chopsticks"
```

We have encountered a similar issue in the past in our own implementation when upgrading from `v1.8` to `v1.9` as a new node function is implemented from that version and forward as shown [here.](https://github.com/paritytech/polkadot-sdk/blob/3c3d6fceb82372a3019b37117aa453d564b212de/templates/parachain/node/src/service.rs#L43C1-L46C4)

**Comment by xlc** [comment]
it needs to be implemented on smoldot first

**Comment by xlc** [comment]
can you provide a wasm and config file so we can later check if the implementation works?

**Comment by metricaez** [comment]
hey! unfortunately I cannot provide the current rpc we are trying with but I could build a simple lean (with no business logic at runtime level) implementation so we can test this case that comes from the node part and share it with you if that makes sense.

**Comment by xlc** [comment]
you can upload a wasm and config file here that can repreoduce this issue
also give https://github.com/AcalaNetwork/chopsticks/pull/755 a try to see if it fixes the issue

**Comment by metricaez** [comment]
Hi, I have built and tested #755 and it does fix the issue, thanks!

On the other hand, I am curious and a little bit confused about the wasm request since the issue comes from the node impl and not the runtime itself. Could you please elaborate ?

**Comment by xlc** [comment]
issue is always coming from wasm because that’s the only thing chopsticks is testing
for this particular case, the wasm is using a new host function that wasn’t implemented by smoldot so I requested smoldot to implement it and have chopsticks updated to include the new host function

**Comment by xlc** [comment]
should be fixed by #755
